File: backend/src/main/java/ru/focsit/backend/dto/FavoriteSongsDto.java
package ru.focsit.backend.dto;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.util.Set;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class FavoriteSongsDto {
    private Long userId;
    private String username;
    private Set<SongDto> favoriteSongs;
}

File: mobile/app/src/main/java/ru/focsit/mobile/data/FavoriteSongsDto.kt
package ru.focsit.mobile.data

/**
 * Данные о любимых песнях пользователя.
 * Используется для хранения информации о песнях, добавленных в избранное.
 *
 * @property userId Идентификатор пользователя, чьи любимые песни представлены.
 * @property username Имя пользователя, чьи любимые песни представлены.
 * @property favoriteSongs Множество песен, добавленных в избранное.
 */
data class FavoriteSongsDto(
    val userId: Long,
    val username: String,
    val favoriteSongs: Set<SongDto>
)